### PR TITLE
qt6: update to 6.7.1

### PR DESCRIPTION
**Pull request by sunderme**
qt-base patches adapted
unnecessary dependencies in qt5compat and qttools removed

### CODE REVIEW

File: mingw-w64-qt6-5compat/PKGBUILD [hunk 1]
```diff
@@ -21,12 +21,10 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-qt6-base"
-             "${MINGW_PACKAGE_PREFIX}-qt6-declarative"
```
Comment (Biswa96) [inline mingw-w64-qt6-5compat/PKGBUILD:24]: Those are required.

```
$ ntldd /ucrt64/share/qt6/qml/Qt5Compat/GraphicalEffects/qtgraphicaleffectsplugin.dll | cut -d '=' -f 1
        KERNEL32.dll
        libstdc++-6.dll
        Qt6Core.dll
        Qt6Qml.dll
```

Qt6Qml.dll is provided by qt6-declarative.

File: mingw-w64-qt6-tools/PKGBUILD [hunk 2]
```diff
@@ -21,15 +21,15 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-qt6-declarative"
-             "${MINGW_PACKAGE_PREFIX}-qt6-activeqt"
```
Comment (Biswa96) [inline mingw-w64-qt6-tools/PKGBUILD:24]: qt6-activeqt is required for qt6-tools according to its build log.

```
-- The following packages have been found:
...
 * Qt6AxContainerTools (required version >= 6.7.0)
 * Qt6AxContainer (required version >= 6.7.0)
...
```

**Comment by lazka** [comment]
qt6-doc:

```
2024-05-30T18:21:03.2316783Z ==> Starting prepare()...
2024-05-30T18:21:03.2330280Z   -> Applying 001-fix-qtdocs-helpers-cmake.patch
2024-05-30T18:21:03.2515539Z patching file cmake/QtDocsHelpers.cmake
2024-05-30T18:21:03.2516621Z Hunk #1 succeeded at 72 (offset 28 lines).
2024-05-30T18:21:03.2577246Z   -> Applying 002-zstd-target.patch
2024-05-30T18:21:03.2732616Z patching file src/designer/src/lib/CMakeLists.txt
2024-05-30T18:21:03.2738171Z Reversed (or previously applied) patch detected!  Skipping patch.
2024-05-30T18:21:03.2739457Z 1 out of 1 hunk ignored -- saving rejects to file src/designer/src/lib/CMakeLists.txt.rej
2024-05-30T18:21:03.3272295Z ==> ERROR: A failure occurred in prepare().
2024-05-30T18:21:03.3533600Z     Aborting...
```

**Comment by Biswa96** [comment]
Please wait qt6 will be updated soon.

**Comment by sunderme** [comment]
I have looked into the issue and apparently it is being worked on for 6.7.2 (changing CMakeList in designer) currently.
The current solution in 6.7.2dev did not resolve the issue.

**Comment by Biswa96** [comment]
Could you explain what the issue is? The patch could be imported from upstream if that does not break current version.

**Comment by sunderme** [comment]
to be honest, I don't understand the issue.
Qt took over the KDE patch for 6.7 (`002-zstd-target.patch::https://invent.kde.org/qt/qt/qttools/-/commit/96306747.patch`)
which basically adds
```
+if(TARGET zstd::libzstd_shared)
+    qt_internal_disable_find_package_global_promotion(zstd::libzstd_shared)
+endif()
```

Currently dev contains: (https://github.com/qt/qttools/blob/dev/src/designer/src/lib/CMakeLists.txt)
```
if(TARGET zstd::libzstd)
    qt_internal_disable_find_package_global_promotion(zstd::libzstd)
endif()
if(TARGET zstd::libzstd_shared)
    qt_internal_disable_find_package_global_promotion(zstd::libzstd_shared)
endif()
if(TARGET zstd::libzstd_static)
    qt_internal_disable_find_package_global_promotion(zstd::libzstd_static)
endif()
if(NOT TARGET WrapZSTD::WrapZSTD)
    qt_find_package(WrapZSTD 1.3
        PROVIDED_TARGETS
            WrapZSTD::WrapZSTD
            zstd::libzstd
            zstd::libzstd_static
            zstd::libzstd_shared
    )
endif()
```
Patching that into 6.7.1 did however not resolve the issue.

**Comment by sunderme** [comment]
the code was added with this comment:
```
[Add the missing global promotion exclution checks for zstd](https://github.com/qt/qttools/commit/604bb3bf2c8e5e26dab678fdf9c730625aa24977)

Add the missing global promotion exclution checks for zstd. Also ensure
the raw zstd targets are promoted to global if they are found for the
first time. Amends https://github.com/qt/qttools/commit/f57d15654b107f70c0ee853b82488ee0c329c3ba Fixes: QTBUG-119469
Change-Id: Ie05ae560243438c6d5d740ca138beff3f4201b80
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
```

**Comment by sunderme** [comment]
maybe the first commit was missing
https://github.com/qt/qttools/commit/f57d15654b107f70c0ee853b82488ee0c329c3ba

**Pull request closed without merging**